pkg/kubernetes/kubernetes.go

```diff
@@ -336,15 +368,43 @@ func CreateCustomResourceDefinition(ctx context.Context, crdName string, crdSing
 
 	// If there is an error, delete the object to keep it clean.
 	if err != nil {
-		log.Infof("Cleanup %q CRD because the CRD created was not successfully established. Error: %+v", crdName, err)
-		deleteErr := apiextensionsClientSet.ApiextensionsV1beta1().CustomResourceDefinitions().Delete(ctx, crdName, metav1.DeleteOptions{})
+		log.Infof("Cleanup %q CRD because the CRD created was not successfully established. Err: %+v", crdName, err)
+		deleteErr := clientSet.ApiextensionsV1beta1().CustomResourceDefinitions().Delete(ctx, crdName, *metav1.NewDeleteOptions(0))
 		if deleteErr != nil {
-			log.Errorf("failed to delete %q CRD with error: %+v", crdName, deleteErr)
+			log.Errorf("Failed to delete %q CRD with err: %+v", crdName, deleteErr)
 		}
 	}
 	return err
 }
 
+// getCRDFromManifest reads a .json/yaml file and returns the CRD in it.
+func getCRDFromManifest(ctx context.Context, fileName string) (*apiextensionsv1beta1.CustomResourceDefinition, error) {
+	var crd apiextensionsv1beta1.CustomResourceDefinition
+	log := logger.GetLogger(ctx)
+
+	fullPath := filepath.Join(manifestPath, fileName)
+	data, err := ioutil.ReadFile(fullPath)
+	if os.IsNotExist(err) {
```

Comment:
We need to check ` os.IsNotExist(err) {` only when `err` is not nil. Can you refactor this part of code?

Reply:
This code deals with the error as well:

```
if os.IsNotExist(err) {
		log.Errorf("Manifest file: %s doesn't exist. Error: %+v", fullPath, err)
		return nil, err
} else if err != nil {
	log.Errorf("Failed to read the manifest file: %s. Error: %+v", fullPath, err)
	return nil, err
}
```

Do you still prefer some refactoring to make it consistent with other code?